language/move-vm/runtime/src/runtime.rs

```diff
@@ -377,19 +377,22 @@ impl VMRuntime {
         for (idx, (arg, arg_type)) in args.into_iter().zip(params).enumerate() {
             match arg_type {
                 Type::MutableReference(inner_t) => {
-                    dummy_locals
-                        .store_loc(idx, self.deserialize_value(&inner_t, arg).unwrap())
-                        .unwrap();
-                    actuals.push(dummy_locals.borrow_loc(idx).unwrap());
+                    if let Err(err) = self.arg_to_actual(idx, arg, &inner_t, &mut dummy_locals, &mut actuals) {
+                        return Err(err.finish(Location::Undefined));
+                    }
                     mut_ref_inputs.push((idx, *inner_t));
                 }
                 Type::Reference(inner_t) => {
-                    dummy_locals
-                        .store_loc(idx, self.deserialize_value(&inner_t, arg).unwrap())
-                        .unwrap();
-                    actuals.push(dummy_locals.borrow_loc(idx).unwrap())
+                    if let Err(err) = self.arg_to_actual(idx, arg, &inner_t, &mut dummy_locals, &mut actuals) {
+                        return Err(err.finish(Location::Undefined));
+                    }
+                }
+                arg_type => {
+                    match self.deserialize_value(&arg_type, arg) {
```

Comment:
Similarly here.

Reply:
with `.map_err()` ?  I think for this one we should maybe keep it as-is, i think the flow is clearest this way.  open to suggestions on how to change it if i'm missing something